homeassistant/components/group/config_flow.py

```diff
@@ -108,14 +110,33 @@ def _set_group_type(user_input: dict[str, Any]) -> dict[str, Any]:
 }
 
 
+def _exclude_self(
+    handler: HelperConfigFlowHandler | HelperOptionsFlowHandler,
+    schema: vol.Schema,
+    options: dict[str, Any],
+) -> vol.Schema:
+    """Exclude own entities from the entity selector."""
+    return exclude_own_entity(handler, schema, CONF_ENTITIES)
+
+
 OPTIONS_FLOW: dict[str, HelperFlowFormStep | HelperFlowMenuStep] = {
     "init": HelperFlowFormStep(None, next_step=choose_options_step),
-    "binary_sensor": HelperFlowFormStep(BINARY_SENSOR_OPTIONS_SCHEMA),
-    "cover": HelperFlowFormStep(basic_group_options_schema("cover")),
-    "fan": HelperFlowFormStep(basic_group_options_schema("fan")),
-    "light": HelperFlowFormStep(LIGHT_OPTIONS_SCHEMA),
-    "media_player": HelperFlowFormStep(basic_group_options_schema("media_player")),
-    "switch": HelperFlowFormStep(SWITCH_OPTIONS_SCHEMA),
+    "binary_sensor": HelperFlowFormStep(
+        BINARY_SENSOR_OPTIONS_SCHEMA, update_form_schema=_exclude_self
```

Comment:
This seems very complicated. I think a better approach would be that the schema can be set to a callable which allows generating the schema on the fly.

Reply:
That's indeed better 👍